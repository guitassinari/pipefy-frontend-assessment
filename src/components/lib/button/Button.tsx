
import React, { HTMLAttributes } from 'react'
import { StyledButton } from './Button.styles'

const Button: React.FC<HTMLAttributes<HTMLButtonElement>> = ({ children, ...rest }) => (
  <StyledButton role="button" {...rest}>
    {children}
  </StyledButton>
)

export default Button