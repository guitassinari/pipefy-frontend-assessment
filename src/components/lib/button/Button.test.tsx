import React from 'react';
import { render, fireEvent } from '@testing-library/react'
import Button from '.';

describe('Modal', () => {
  it('matches snapshot', () => {
    const component = render(<Button>Modal test</Button>)

    expect(component.asFragment()).toMatchSnapshot()
  })

  it('renders its received children', () => {
    const component = render(<Button>Modal test</Button>)

    expect(component.getByText('Modal test')).toBeInTheDocument()
  })

  describe('when receives a onClick', () => {
      it('executes the onClick callback whenever it is clicked', () => {
        const onClickCallback = jest.fn()
        const component = render(<Button onClick={onClickCallback}>Modal test</Button>)

        fireEvent.click(component.getByRole("button"))

        expect(onClickCallback).toHaveBeenCalled()
      })
  })
})
