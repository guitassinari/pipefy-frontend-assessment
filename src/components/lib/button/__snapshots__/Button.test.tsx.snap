// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL hOFFMg"
    role="button"
  >
    Modal test
  </button>
</DocumentFragment>
`;
